From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: docker-mirakc-edcb <docker-mirakc-edcb@localhost>
Date: Mon, 5 Oct 2026 02:41:45 +0900
Subject: [PATCH 1/3] Resolve SERVER_HOST on every connect

SERVER_HOST was passed to inet_addr(), so only an IPv4 address worked.
Resolve it with getaddrinfo() (IPv4) on every connect instead of once,
so that a host name such as a Docker service name works and a changed
address is followed without reloading the driver. A failed lookup is
reported and handled as a failed connect.
---
 src/http_proc.hpp | 48 +++++++++++++++++++++++------------------------
 1 file changed, 24 insertions(+), 24 deletions(-)

diff --git a/src/http_proc.hpp b/src/http_proc.hpp
index f865efe..91491c6 100644
--- a/src/http_proc.hpp
+++ b/src/http_proc.hpp
@@ -54,12 +54,9 @@ public:
 class MirakcConnectHttp : public MirakcConnectBase
 {
 private:
-#if 1
-	struct sockaddr_in s_addr;
-#else
-	struct addrinfo hints;
-	struct addrinfo *addr;
-#endif
+	// resolved on every connect() so that a changed address is followed
+	char host[ 256 ];
+	unsigned short port;
 	
 public:
 	MirakcConnectHttp( char *host, short port );
@@ -82,34 +79,37 @@ public:
 
 MirakcConnectHttp::MirakcConnectHttp( char *host, short port )
 {
-#if 1
-	s_addr.sin_addr.s_addr = inet_addr( host );
-	s_addr.sin_port        = htons( port );
-	s_addr.sin_family      = AF_INET;
-#endif
+	::snprintf( this->host, sizeof( this->host ), "%s", host );
+	this->port = (unsigned short)port;
+}
+
+
+int MirakcConnectHttp::connect()
+{
+	int ret;
+	struct addrinfo hints;
+	struct addrinfo *addr;
+	char service[ 8 ];
 
-#if 0
+	// an IP address or a host name (IPv4 only)
 	memset( &hints, 0, sizeof(hints) );
 	hints.ai_socktype = SOCK_STREAM;
 	hints.ai_family = AF_INET;
+	::snprintf( service, sizeof( service ), "%u", (unsigned int)port );
 
-	int ret; 
-	
-	ret = getaddrinfo( host, NULL, &hints, &addr );
+	ret = getaddrinfo( host, service, &hints, &addr );
 	if( ret != 0 ) {
-		// error
+		ERROR_OUTPUT( "cannot resolve %s (%s)", host, gai_strerror( ret ) );
+		s = -1;
+		errno = EHOSTUNREACH;
+		return -1;
 	}
-#endif
-}
 
-
-int MirakcConnectHttp::connect()
-{
-	int ret;
-	
 	s = socket(AF_INET, SOCK_STREAM, 0);
 
-	ret = ::connect( s, (sockaddr *)&s_addr, sizeof( s_addr ) );
+	ret = ::connect( s, addr->ai_addr, addr->ai_addrlen );
+
+	freeaddrinfo( addr );
 	
 	return ret;
 }
